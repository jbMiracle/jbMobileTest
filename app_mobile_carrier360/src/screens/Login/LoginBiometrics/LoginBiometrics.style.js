import C360Colors from '../../../themes/360Colors';

export default {
  container: {
    alignItems: 'flex-start',
    flex: 1,
    flexDirection: 'row',
  },
  subContainer: {
    alignItems: 'center',
    flex: 1,
    paddingHorizontal: 25,
  },
  H3: {
    color: C360Colors.White,
    fontWeight: 'bold',
    marginBottom: 15,
    textAlign: 'center',
  },
  desc: {
    color: C360Colors.White,
    textAlign: 'center',
  },
  btnText: {
    color: C360Colors.Blue900,
    fontWeight: '600',
  },
  logo: {
    height: 160,
    marginVertical: 30,
    width: 160,
  },
  biometricImage: {
    height: 140,
    marginVertical: 30,
    width: 140,
  },
};
