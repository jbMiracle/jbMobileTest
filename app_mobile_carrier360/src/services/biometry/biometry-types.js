export default {
  TouchID: 'touch_id',
  FaceID: 'face_id',
};
