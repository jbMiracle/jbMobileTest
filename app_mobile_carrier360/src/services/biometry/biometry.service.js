import {
  Platform,
  // Alert,
} from 'react-native';
import TouchID from 'react-native-touch-id';
import BiometryTypes from './biometry-types';

export async function queryBiometricSupport() {
  try {
    // On Android: boolean
    // On iOS: string of touch/face
    const supportResult = await TouchID.isSupported();
    const supportedType = Platform.select({
      android: supportResult ? BiometryTypes.TouchID : null,
      ios: BiometryTypes[supportResult],
    });
    return {
      isSupported: !!supportResult,
      supportedType,
    };
  } catch (err) {
    return {
      isSupported: false,
      supportedType: null,
    };
  }
}

// const alertResult = result => new Promise((resolve) => {
//   Alert.alert(
//     'Biometric Authentication Result!',
//     result,
//     [
//       { text: 'OK', onPress: resolve },
//     ],
//     { cancelable: false }
//   );
// });

export async function authenticate(message = 'Pretty Please!') {
  try {
    const result = await TouchID.authenticate(message);
    // await alertResult(JSON.stringify(result));
    return result;
  } catch (err) {
    // User cancelling dialog brings us here.. now what?
    // On failed touch attempt:
    //   Android immediately throws failure
    //   iOS gives the user another chance before yielding a result/failure
    //     The user can try again or tap 'use password' which then throws back "LAErrorUserFallback"(iOS) / "Touch ID Error"(Android)
    // alertResult(JSON.stringify(err, null, 2));
    return false;
  }
}
