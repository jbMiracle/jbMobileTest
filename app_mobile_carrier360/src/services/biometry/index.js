import BiometryTypes from './biometry-types';

export { BiometryTypes };
export * from './biometry.service';
