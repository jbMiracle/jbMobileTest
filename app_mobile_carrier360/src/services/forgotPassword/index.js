import sendForgotPassword from './forgotPassword.service';

export default sendForgotPassword;
