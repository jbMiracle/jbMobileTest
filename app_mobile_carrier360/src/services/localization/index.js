import strings from './localization.service';

export default strings;
