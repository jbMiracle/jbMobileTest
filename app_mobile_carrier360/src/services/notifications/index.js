export * from './notifications.service';
export { ActionCategoryIds } from './actionCategories';
