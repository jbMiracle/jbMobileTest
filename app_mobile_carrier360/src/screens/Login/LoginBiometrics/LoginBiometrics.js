import React, { Component } from 'react';
import { Image } from 'react-native';
import { Text, Button } from 'native-base';
import styles from './LoginBiometrics.style';
import localizedString from '../../../services/localization';
import { authenticate as biometricAuthenticate, BiometryTypes } from '../../../services/biometry';
import C360LogoContainer from '../../../components/C360LogoContainer/C360LogoContainer';

const fingerprintImage = require('../../../assets/images/fingerprint.png');
const faceImage = require('../../../assets/images/faceIdLogoLarge.png');

const BiometryImages = {
  [BiometryTypes.TouchID]: fingerprintImage,
  [BiometryTypes.FaceID]: faceImage,
};

const loginTextOptions = { scope: 'Login' };
const loginString = key => localizedString(key, loginTextOptions);

class LoginBiometrics extends Component {
  touchIdDialog() {
    biometricAuthenticate().then(result => this.props.authenticated(result));
  }

  renderButton(isSetup, labelKey) {
    if (!isSetup) {
      return (
        <Button
          testID='setup_touch_id'
          block
          large
          primaryDark
          onPress={ () => this.props.setupBiometricLogin() }
        >
          <Text>{loginString(labelKey)}</Text>
        </Button>
      );
    } else {
      this.touchIdDialog();
      return null;
    }
  }

  render() {
    const { isBiometricLoginSetup, supportedBiometryType } = this.props;
    return (
      <C360LogoContainer
        headerText={ loginString(`login_${supportedBiometryType}`) }
        descriptionText={ loginString(`login_${supportedBiometryType}_desc`) }
      >
        <Image style={ styles.biometricImage } source={ BiometryImages[supportedBiometryType] } />
        {this.renderButton(isBiometricLoginSetup, `setup_${supportedBiometryType}`)}
      </C360LogoContainer>
    );
  }
}

export default LoginBiometrics;
