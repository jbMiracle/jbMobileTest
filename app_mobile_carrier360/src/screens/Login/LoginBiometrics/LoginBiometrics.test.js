import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import LoginBiometrics from './LoginBiometrics';
import { BiometryTypes } from '../../../services/biometry';

it('can render a menu list component', () => {
  const renderer = new ShallowRenderer();

  const props = {
    isBiometricLoginSupported: true,
    supportedBiometryType: BiometryTypes.FaceID,
    isBiometricLoginSetup: false,
  };

  renderer.render(<LoginBiometrics { ...props } />);

  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
