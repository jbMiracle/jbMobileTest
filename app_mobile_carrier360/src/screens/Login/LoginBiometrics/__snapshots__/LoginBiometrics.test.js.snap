// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render a menu list component 1`] = `
<C360LogoContainer
  descriptionText="You need to setup Face ID on your device that you will use to log in to J.B. Hunt Carrier 360."
  headerText="Login using Face ID"
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/faceIdLogoLarge.png",
      }
    }
    style={
      Object {
        "height": 140,
        "marginVertical": 30,
        "width": 140,
      }
    }
  />
  <Styled(Button)
    block={true}
    large={true}
    onPress={[Function]}
    primaryDark={true}
    testID="setup_touch_id"
    virtual={undefined}
  >
    <Styled(Text)
      virtual={undefined}
    >
      Setup Face ID
    </Styled(Text)>
  </Styled(Button)>
</C360LogoContainer>
`;
